system/services/security/PermissionService.cfc

```diff
@@ -52,6 +52,33 @@ component displayName="Admin permissions service" {
 		return _getRoles().keyArray();
 	}
 
+	public struct function listRolesWithGroup() {
+		var cacheKey  = "admin-roles-with-group";
+		var fromCache = _getCacheProvider().get( cacheKey );
+
+		if ( !IsNull( local.fromCache ) ) {
+			return fromCache;
+		}
+
+		var grouped = {};
+		var roles   = listRoles();
+
+		for ( var role in roles ) {
+			var roleGroup = $translateResource( uri="roles:#role#.group", defaultValue="" );
```

Comment:
We can't cache translations unless we include the current language in the cache key :). Otherwise multilingual admin fails.

Current language can be got from i18n service I believe.